src/v/cluster/controller.cc

```diff
@@ -190,23 +191,33 @@ ss::future<> controller::start(
     producer_id_recovery,
   ss::shared_ptr<cluster::cloud_metadata::offsets_recovery_requestor>
     offsets_recovery) {
-    auto initial_raft0_brokers = discovery.founding_brokers();
+    auto founding_brokers = discovery.founding_brokers();
     std::vector<model::node_id> seed_nodes;
-    seed_nodes.reserve(initial_raft0_brokers.size());
+    seed_nodes.reserve(founding_brokers.size());
     std::transform(
-      initial_raft0_brokers.cbegin(),
-      initial_raft0_brokers.cend(),
+      founding_brokers.cbegin(),
+      founding_brokers.cend(),
       std::back_inserter(seed_nodes),
       [](const model::broker& b) { return b.id(); });
+    std::vector<raft::vnode> initial_raft0_nodes;
+    initial_raft0_nodes.reserve(founding_brokers.size());
+    std::transform(
+      founding_brokers.cbegin(),
+      founding_brokers.cend(),
+      std::back_inserter(initial_raft0_nodes),
+      [](const model::broker& b) {
+          return raft::vnode(b.id(), model::revision_id(0));
+      });
 
     return validate_configuration_invariants()
-      .then([this, initial_raft0_brokers]() mutable {
-          return create_raft0(
-            _partition_manager,
-            _shard_table,
-            config::node().data_directory().as_sstring(),
-            std::move(initial_raft0_brokers));
-      })
+      .then(
+        [this, initial_raft0_nodes = std::move(initial_raft0_nodes)]() mutable {
+            return create_raft0(
+              _partition_manager,
+              _shard_table,
+              config::node().data_directory().as_sstring(),
+              std::move(initial_raft0_nodes));
```

Comment:
I wonder how this works - here raft0 is started while we don't yet have any info about brokers corresponding to these vnodes in the members table. Shouldn't it lead to e.g. node rpc clients not being found?